File: pset1/greedy.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>
float money;

int main(void)
{
    do
    {
    printf("O hai! How much change is owed?\n");
    money = get_float();
    }
    while (money < 0);
    int r = 0;
    int change = roundf (money * 100);
    
    int quarter = change / 25;
    r = change % 25;
    int dime = r / 10;
    r = r % 10;
    int nickel = r / 5;
    r = r % 5;
    int penny = r / 1;
    int output = (quarter + dime + nickel + penny);
    printf("%i\n", output);
    
}

File: pset6/greedy.py
import cs50
#prompt user input
print("O hai! How much change is owed?")
while True:
    m = cs50.get_float()
    if m > 0:
        break
#
c = round(m * 100)
#divide by quarter value
q = c // 25
#get remainder
r = c % 25
#divide by dime value
d = r // 10
#get remainder
r = r % 10
#divide by nickel value
n = r //5
#get remainder
r = r % 5
#divide by penny value
p = r // 1
#add coins count for each value
o = q + d + n + p
#print result
print(o)